app/models/miq_queue.rb

```diff
@@ -179,6 +179,12 @@ def self.put_deprecated(*args)
     put(*args)
   end
 
+  # Place holder for deprecated put_unless_exists call
+  # which might not be needed in the new architecture
+  def self.put_unless_exists_deprecated(*args)
```

Comment:
@jrafanie Agreed.

You're trying to whitelist certain uses of this method and deprecate others but in doing so you're invalidating the concept of 'deprecation'. I would either:

* Leave this alone and deprecate when...well...the method actually can be considered deprecated
* Deprecate it as normal and be ok with deprecation warnings until it can be replaced fully. 

Unless I was looking at this PR I would read `put_unless_exists_deprecated` and be very, very confused.